src/state/application/updater.ts

```diff
@@ -20,6 +22,40 @@ function useQueryCacheInvalidator() {
   }, [chainId, dispatch])
 }
 
+const NETWORK_HEALTH_CHECK_MS = ms`15s`
+const DEFAULT_MS_BEFORE_WARNING_WAIT = ms`10m`
+interface UseBlockWarningTimerArgs {
+  chainId: number | undefined
+  dispatch: (action: any) => void
+  msSinceLastBlock: number
+  setMsSinceLastBlock: (n: number) => void
+}
+
+function useBlockWarningTimer({ chainId, dispatch, msSinceLastBlock, setMsSinceLastBlock }: UseBlockWarningTimerArgs) {
+  const chainConnectivityWarningActive = useAppSelector((state) => state.application.chainConnectivityWarning)
+  const timeout = useRef<NodeJS.Timeout>()
+  useEffect(() => {
+    const waitMsBeforeWarning =
+      (chainId ? CHAIN_INFO[chainId]?.blockWaitMsBeforeWarning : DEFAULT_MS_BEFORE_WARNING_WAIT) ||
```

Comment:
`??` is preferred over `||` here I think in case `CHAIN_INFO[chainId]?.blockWaitMsBeforeWarning` is 0.